### DISCUSSION TITLE
Support for encrypted input streams?

### DISCUSSION THREAD

**Discussion by nikoschule** [OvenMediaLabs/OvenMediaEngine#1191] [Help] [answered]
What is the current status regarding the **support for encrypted input streams**? And related plans for the future?

According to my investigations, the current situation with different protocols is as follows:

**SRT:**
The protocol itself supports end-to-end encryption (AES 128/256) but it seems that this is currently not supported by OvenMediaEngine. Correct? Any idea about how complex this would be to implement?

**WebRTC:**
WebRTC input/signaling is natively transferred through TLS/HTTPS. Does this mean that also the actual video payload is encrypted? So can we say that WebRTC streams handled by OvenMediaEngine are fully encrypted? (I'm not fully familiar with all the low-level details about how WebRTC works.)

**RTMPS:**
RTMPS provider support has been mentioned in the old roadmap (#548 , #315 ) but not in the new one (#1064 ). Any updates on this?

Personally I feel that RTMPS could be of low priority but full encryption support for SRT and WebRTC would be important as they seem to be the most important protocols for the future.

**Comment by bchah** [comment]
RTMPS is possible via an easy workaround detailed in #1133, however the workaround is having issues as of OME 0.15.9 (#1188).

WebRTC inputs could be considered “end-to-end encrypted” since they move over TLS. But somebody smarter than me needs to confirm your question about the video payload itself.

And I agree SRT encryption would be great! Perhaps with the ability to set some XML flags to specify 128 or 256 bit plus a passphrase for the server.

**Comment by getroot** [comment, marked as answer]
WebRTC is encrypted with DTLS and SRTP. OME works like that, of course.

OvenMediaEngine has only library integration for SRT. We haven't finished testing yet for options like passphrase.

**Comment by nikoschule** [comment]
Big thanks for these clarifications (and closing the mentioned issue #1188 related to RTMPS workaround) @getroot and @bchah !

The full encryption support for SRT would still be great at some point in the future.
